pkg/iter/sc_finder.go

```diff
@@ -194,6 +195,15 @@ func (m *SubclusterFinder) buildObjList(ctx context.Context, list client.ObjectL
 	return meta.SetList(list, rawObjs)
 }
 
+// shouldSkipBasedOnSandboxState returns true if the object whose labels
+// is passed does not belong to the given sandbox or main cluster
+func shouldSkipBasedOnSandboxState(l map[string]string, sandbox string) bool {
+	if sandbox == vapi.MainCluster {
+		return l[vmeta.SandboxNameLabel] != vapi.MainCluster
+	}
+	return l[vmeta.SandboxNameLabel] != sandbox
```

Comment:
Couldn't you just simply it this way?
```suggestion
	return l[vmeta.SandboxNameLabel] != sandbox
```